src/uri-template.js

```diff
@@ -16,7 +16,13 @@ export default function buildUriTemplate(basePath, href, pathObjectParameters =
     const parameterNames = _.unique([].concat(pathObjectParameterNames, queryParameterNames));
     const parameterNamesString = parameterNames.length ? `{?${parameterNames.join(',')}}` : '';
 
-    return `${basePath}${href}${parameterNamesString}`;
+    const full = `${basePath}${href}${parameterNamesString}`;
+
+    // Before returning, we replace instances of `-` with `%2d`, but only when
+    // they occur inside of a template variable.
+    return full.replace(/\{.*?\}/g, (match) => {
```

Comment:
My question I had before was, should this be done for the full URI? Don't dashes only need to be replaced in the variables and not in the `basePath` and `href`?


In other words, in your tests, if you set the `basePath` to `/my-api`, it would convert that dash as well. Thoughts?